src/types.mts

```diff
@@ -0,0 +1,112 @@
+// Local Types
+type TConfigurationObjectValue = string | RegExp | object | boolean | typeof Function;
+
+type TCustomConfiguration = Omit<IConfigurationInternal, 'accessToken'> & {
+  onUnhandledError?: string;
+  onUnhandledPromiseRejection?: string;
```

Comment:
Nevermind, the `hasConfigurationInPayload` test breaks when you remove this

Reply:
Not sure why `TSerializedObject` isn't picking those up properly, seems like it should. But these are needed to eliminate type errors.

Reply:
It looks like `TSerializedObject` is wrong indeed 🤔 

Somehow [this version](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYgrgOwMbAJYHsEEEDOBlYAJ1QQHMAeAFQD4oBeKSqCAD2AgQBMdZEUMEUAPxQcREqSgAuRgG4AUKEhQAShFIBRFmFwFiZKrQZNW7LjzWbtw0eLLS58+SXaEAZgEMk0GB4DWEADyAEYAVhAoUADe8lBxUOxiMp4ANjjQAD68yGiYCvEJEGIATDLwOQL58YnAAMzJHmmZqupaYFVxNQAsMmL6pB2FYgCsMghwALbBEIRQWX0Sc9n8mEuWbQoAvgqK4NCUeDOojagAXhCcIeEoVMxsHNxQ6GERwEbRsVAA2gEgUCRQX7oNyMAC6MnW2l0dgo5RW2HwMKoPwgIFB1GoWx2u2Uh2IJ3OnF8ASur3ojDxxxSZwupJuxKCLxQmKAA) works, it looks pretty similar but I am guessing having an intermediate type makes the `extends` apply on each member of the union instead of on the union as a whole 🤷‍♂️